contributed_definitions/nyaml/NXmpes.yaml

```diff
@@ -288,34 +288,129 @@ NXmpes(NXobject):
       (NXmanipulator):
         exists: optional
         doc: |
-          Manipulator for positioning of the sample.
-        sample_temperature(NX_FLOAT):
+          Manipulator for positioning of the sample.      
+        temperature_sensor(NXsensor):
           exists: recommended
-          unit: NX_TEMPERATURE
-        drain_current(NX_FLOAT):
+          doc: Temperature sensor measuring the sample temperature.
+          name:
+            exists: recommended
+          measurement:
+            enumeration: [temperature]
+          type:
+            exists: recommended
+          value(NX_FLOAT):
+            exists: recommended
+            unit: NX_TEMPERATURE
+            doc: |
+              In the case of a fixed temperature measurement, this is the scalar temperature measured
+              by the sample temperature sensor.
+          value_log(NXlog):
+            exists: optional
+            value(NX_NUMBER):
+              doc: |
+                In the case of an experiment in which the temperature is changed and recorded,
+                this is an array of length m of temperatures.
+              unit: NX_TEMPERATURE
+        sample_heater(NXactuator):
           exists: recommended
-          unit: NX_CURRENT
-        sample_bias(NX_FLOAT):
+          doc: Device to heat the sample.
+          name:
+            exists: recommended
+          actuation:
+            enumeration: [temperature]
+          type:
+            exists: recommended
+          setpoint_log(NXlog):
+            exists: recommended
+            value:
+              unit: NX_TEMPERATURE
+          output_log(NXlog):
+            value:
+              unit: NX_TEMPERATURE
```

Comment:
I think the main blocker with this PR is that `NXsensor` is somewhat poorly designed for this use case, especially the distinction between `value` and `value_log`. This leads to the problems outlined above by @rettigl , i.e., deprecated fields, unnecessary limitations, etc. Ideally, `NXsensor` should be restructured to be in line with `NXactuator`. You can see the difference in using the two base classes here. However, since `NXsensor` is an existing base class and shall not be changed to not break other NeXus use cases, we need to use the workarounds in ´NXmpes/NXmanipulator´.